cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java

```diff
@@ -0,0 +1,184 @@
+            }
+        };
+
+        abstract Map<String, String> create();
+    }
+
+    public OrderedProperties(Ordering ordering) {
+        pairs = ordering.create();
+    }
+
+    /**
+     * Filter key/value pairs
+     *
+     * @param filter The predicate which determines wheather key/value pair will be retained.  The
+     * predicate is invoked with the key.  The pair is removed if the predicate returns true.
+     * @return this instance for chaining
+     */
+    public OrderedProperties filter(Predicate<String> filter) {
+        for (Iterator<Entry<String, String>> it = pairs.entrySet().iterator(); it.hasNext(); ) {
+            if (filter.test(it.next().getKey())) {
+                it.remove();
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Add Key/Value pairs
+     *
+     * @param additionalPairs Key/Value pairs to add to current set (may be null)
+     * @return this instance for chaining
+     */
+    public OrderedProperties addMapPairs(Map<String, String> additionalPairs) {
+        pairs.putAll(additionalPairs);
+        return this;
+    }
+
+    /**
+     * Converts the values from the Iterable into String values and adds the pairs to the current set.
+     *
+     * @param iterable Key/Value pairs
+     * @return this instance for chaining
+     * @throws InvalidConfigParameterException if value is null or is of an unsupported type
+     */
+    public OrderedProperties addIterablePairs(Iterable<? extends Map.Entry<String, ? extends Object>> iterable) {
+        for (Map.Entry<String, ? extends Object> entry : iterable) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+
+            if (value instanceof String) {
+                pairs.put(key, (String) value);
+            } else if (value instanceof Integer || value instanceof Long || value instanceof Boolean
+                || value instanceof Double || value instanceof Float) {
+                pairs.put(key, String.valueOf(value));
+            } else if (value == null) {
+                throw new InvalidConfigParameterException(key, "A null value is not allowed for this key");
+            } else {
+                throw new InvalidConfigParameterException(key,
+                    " - Unsupported type " + value.getClass() + " in configuration for this key");
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Parse key/value pairs and add to the current pair set.
+     * N.B. values containing a newline or other control characters will probably cause problems!
+     * @param keyValuePairs Pairs in key=value format, pairs separated by newlines
+     * @throws InvalidConfigParameterException if value is null
+     * @return this instance for chaining
+     */
+    public OrderedProperties addStringPairs(String keyValuePairs) {
+        try (LineNumberReader lnr = new LineNumberReader(new StringReader(keyValuePairs))) {
```

Comment:
If you don't need to report the line number, won't a `BufferedReader` suffice?

Reply:
yes, will change.